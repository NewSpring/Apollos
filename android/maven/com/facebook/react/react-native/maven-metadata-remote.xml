<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>28.0.0</release>
    <versions>
      <version>28.0.0</version>
    </versions>
    <lastUpdated>20180702224301</lastUpdated>
  </versioning>
</metadata>
