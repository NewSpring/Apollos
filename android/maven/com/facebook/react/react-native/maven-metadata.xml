<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>25.0.0</release>
    <versions>
      <version>25.0.0</version>
    </versions>
    <lastUpdated>20180118183717</lastUpdated>
  </versioning>
</metadata>
