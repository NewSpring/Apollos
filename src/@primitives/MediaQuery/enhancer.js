import { branch, compose } from 'recompose';
import { find } from 'lodash';
import withWindow from './withWindow';
import withTheme from '../withTheme';

// Make it easy to do breakpoint-based component branching, a-la recompose's branch method,
// except the first argument is an object to test against versus a function.
// This object mirrors the props supported by <MediaQuery> and the intention would be to
// use it like a mediaQuery:
// import { enhancer as mediaQuery } from 'MediaQuery';
// mediaQuery(
//   ({ xs, lg }) => ({ minWidth: xs, maxWidth: lg }),
//   HigherOrderComponent,
//   ?HigherOrderComponent
// ): HigherOrderComponent
export default (getMediaQuery, ...args) => compose(
  withTheme(({ theme: { breakpoints = {} } = {} }) => ({ breakpoints })),
  withWindow,
  branch(({ breakpoints, window: { width, height }, ...ownProps }) => {
    const mediaQuery = getMediaQuery({ ...breakpoints, width, height }, ownProps);

    // We should only interpret `mediaQuery` if we have an object.
    // This allows the function you pass in to utilize more complex query logic, like:
    // mediaQuery(({ sm, lg, width }) => width < sm || width > lg, styles)
    // mediaQuery(({ sm }, ownProps) => ownProps.size < sm, styles)
    if (typeof mediaQuery !== 'object') {
      return !!mediaQuery;
    }

    // This logic is a litle goofy, but here's the intention:
    // We want to make sure that all terms in the provided media query are true.
    // Instead of iterating every term, we can exit if we come across a falsey term.
    // So we use `find` to look for a query that doesn't match the current window.
    // We then invert the result of `find` and return it.
    // TODO: refactor so we don't need 5 lines of explanation
    const mediaQueryMatches = !find(mediaQuery, (selector, query) => {
      switch (query) {
        case 'maxWidth': return !(width < selector);
        case 'minWidth': return !(width > selector);
        case 'maxHeight': return !(height < selector);
        case 'minHeight': return !(height > selector);
        case 'minDeviceAspectRatio': return !(width / height > selector);
        case 'maxDeviceAspectRatio': return !(width / height < selector);
        default: return true;
      }
    });

    return mediaQueryMatches;
  }, ...args),
);
