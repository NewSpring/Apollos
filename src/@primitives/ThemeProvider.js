import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME, THEME_PROPS } from './constants';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: Object.assign({}, DEFAULT_THEME, this.props.theme),
  })

  render() {
    return Children.only(this.props.children);
  }
}
