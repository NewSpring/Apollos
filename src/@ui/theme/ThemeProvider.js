import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import createTheme, { THEME_PROPS } from './createTheme';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    themeInput: {},
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: createTheme(this.props.themeInput),
    themeInput: this.props.themeInput,
  })

  render() {
    return Children.only(this.props.children);
  }
}
