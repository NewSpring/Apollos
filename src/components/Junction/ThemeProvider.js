import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME } from './constants';

export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  getChildContext = () => ({
    theme: this.props.theme,
  })

  render() {
    return Children.only(this.props.children);
  }
}
