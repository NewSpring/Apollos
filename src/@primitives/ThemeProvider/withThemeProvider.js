import { mapProps, withContext, compose } from 'recompose';
import PropTypes from 'prop-types';
import { THEME_PROPS } from '@primitives/constants';
import withTheme from '@primitives/withTheme';

// HOC to help with extending theme.
// Use to over-ride theme context for child components
// Does not change or modify props of the component you're enhancing
const withThemeProvider = themeInput => compose(
  mapProps(props => ({ ownProps: props })),
  withTheme(),
  withContext({ theme: PropTypes.shape(THEME_PROPS) }, ({ theme, ownProps }) => {
    let composedTheme = Object.assign({}, theme, ownProps.theme || {});

    let themeInputAsObject = themeInput;
    if (typeof themeInput === 'function') themeInputAsObject = themeInput({ ...ownProps, theme: composedTheme });

    composedTheme = Object.assign({}, composedTheme, themeInputAsObject);

    return ({
      theme: composedTheme,
    });
  }),
  mapProps(({ ownProps }) => ownProps),
);

export default withThemeProvider;
