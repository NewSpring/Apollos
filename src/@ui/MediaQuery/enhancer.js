import { branch, compose, mapProps } from 'recompose';
import { every } from 'lodash';
import { withTheme } from '@ui/theme';

import withWindow from './withWindow';
import queryMatcher from './queryMatcher';

// Make it easy to do breakpoint-based component branching, a-la recompose's branch method.
// The "test" function should return an object that mirrors what you'd do in a css mediaQuery:
// import { enhancer as mediaQuery } from 'MediaQuery';
// mediaQuery(
//   ({ xs, lg }) => ({ minWidth: xs, maxWidth: lg }),
//   HigherOrderComponent,
//   ?HigherOrderComponent
// ): HigherOrderComponent
const enhancer = (getMediaQuery, ...args) => compose(
  withTheme(({ theme: { breakpoints = {} } = {} }) => ({ breakpoints })),
  withWindow,
  branch(({ breakpoints, window: { width, height }, ...ownProps }) => {
    const mediaQuery = getMediaQuery({ ...breakpoints, width, height }, ownProps);

    // We should only interpret `mediaQuery` if we have an object.
    // This allows the function you pass in to utilize more complex query logic, like:
    // mediaQuery(({ sm, lg, width }) => width < sm || width > lg, styles)
    // mediaQuery(({ sm }, ownProps) => ownProps.size < sm, styles)
    if (typeof mediaQuery !== 'object') {
      return mediaQuery;
    }

    return every(mediaQuery, queryMatcher({ width, height }));
  }, ...args),

  // clean up props
  mapProps(({ breakpoints, window, ...ownProps }) => ownProps),
);

export default enhancer;
