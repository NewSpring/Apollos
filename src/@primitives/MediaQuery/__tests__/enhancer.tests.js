import React from 'react';
import renderer from 'react-test-renderer';
import { View, Dimensions } from 'react-native';
import ThemeProvider from '@primitives/ThemeProvider';
import enhancer from '../enhancer';

const TestHOC = mock => (Component) => {
  mock();
  return Component;
};

describe('The mediaQuery enhancer', () => {
  it('uses the first HOC when truthful', () => {
    const firstMock = jest.fn();
    const secondMock = jest.fn();

    const Component = enhancer(() => true, TestHOC(firstMock), TestHOC(secondMock))(View);
    const tree = renderer.create(
      <ThemeProvider>
        <Component />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
    expect(firstMock.mock.calls.length).toBe(1);
    expect(secondMock.mock.calls.length).toBe(0);
  });
  it('uses the second HOC when falsey', () => {
    const firstMock = jest.fn();
    const secondMock = jest.fn();

    const Component = enhancer(() => false, TestHOC(firstMock), TestHOC(secondMock))(View);
    const tree = renderer.create(
      <ThemeProvider>
        <Component />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
    expect(firstMock.mock.calls.length).toBe(0);
    expect(secondMock.mock.calls.length).toBe(1);
  });

  const testWindow = Dimensions.get('window');
  it('works with a simple media query', () => {
    const firstMock = jest.fn();

    const Component = enhancer(
      () => ({ maxWidth: testWindow.width + 1 }),
      TestHOC(firstMock),
    )(View);
    const tree = renderer.create(
      <ThemeProvider>
        <Component />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
    expect(firstMock.mock.calls.length).toBe(1);
  });
  it('works with a complex media query', () => {
    const firstMock = jest.fn();

    const Component = enhancer(
      () => ({
        maxWidth: testWindow.width + 1,
        minWidth: testWindow.width - 1,
        maxHeight: testWindow.height + 1,
        minHeight: testWindow.height - 1,
        minDeviceAspectRatio: (testWindow.width / testWindow.height) - 1,
        maxDeviceAspectRatio: (testWindow.width / testWindow.height) + 1,
      }),
      TestHOC(firstMock),
    )(View);
    const tree = renderer.create(
      <ThemeProvider>
        <Component />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
    expect(firstMock.mock.calls.length).toBe(1);
  });
  it('passes in breakpoints to the test function', () => {
    const firstMock = jest.fn();

    const Component = enhancer(
      (breakpoints) => {
        expect(Object.keys(breakpoints)).toEqual(expect.arrayContaining(['xs', 'sm', 'md', 'lg']));
        return true;
      },
      TestHOC(firstMock),
    )(View);
    const tree = renderer.create(
      <ThemeProvider>
        <Component />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
    expect(firstMock.mock.calls.length).toBe(1);
  });
});
