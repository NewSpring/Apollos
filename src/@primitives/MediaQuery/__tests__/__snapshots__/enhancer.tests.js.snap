// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The mediaQuery enhancer passes in breakpoints to the test function 1`] = `<View />`;

exports[`The mediaQuery enhancer passes in window width and height to the test function 1`] = `<View />`;

exports[`The mediaQuery enhancer uses second HOC when provided with a complex media query that doesnt match 1`] = `<View />`;

exports[`The mediaQuery enhancer uses the first HOC (and renders the root View) with a passing media query 1`] = `<View />`;

exports[`The mediaQuery enhancer uses the second HOC (and renders the root View) with a failing media query 1`] = `<View />`;

exports[`The mediaQuery enhancer works with a complex media query 1`] = `<View />`;

exports[`The mediaQuery enhancer works with a simple media query 1`] = `<View />`;
